tiledb/sm/query/ast/query_ast.cc

```diff
@@ -45,6 +45,89 @@ static inline bool supported_string_type(Datatype type) {
       type == Datatype::STRING_UTF8);
 }
 
+ASTNodeVal::ASTNodeVal(
+    const std::string& field_name,
+    const void* const data,
+    uint64_t data_size,
+    const void* const offsets,
+    uint64_t offsets_size,
+    const QueryConditionOp op,
+    bool use_enumeration)
+    : field_name_(field_name)
+    , is_null_(false)
+    , op_(op)
+    , use_enumeration_(use_enumeration) {
+  if (data == nullptr) {
+    throw std::invalid_argument(
+        "ASTNodeVal set membership data must not be nullptr");
+  }
+
+  if (data_size == 0) {
```

Comment:
Could you have a set of empty strings?

Reply:
Super good catch. Fixed and added a test.